onnxruntime/core/providers/tensorrt/tensorrt_execution_provider_utils.h

```diff
@@ -170,134 +170,126 @@ std::string GetCachePath(const std::string& root, const std::string& name) {
-    for (const auto* node_arg : main_graph.GetInputsIncludingInitializers()) {
-      hash_str(node_arg->Name());
-    }
+  // fingerprint the main graph by hashing graph inputs
+  for (const auto* node_arg : main_graph.GetInputsIncludingInitializers()) {
+    hash_str(node_arg->Name());
+  }
 
-    // hashing output of each node
-    const int number_of_ort_nodes = graph_viewer.NumberOfNodes();
-    std::vector<size_t> nodes_vector(number_of_ort_nodes);
-    std::iota(std::begin(nodes_vector), std::end(nodes_vector), 0);
-    const std::vector<NodeIndex>& node_index = graph_viewer.GetNodesInTopologicalOrder();
-    for (const auto& index : nodes_vector) {
-      const auto& node = graph_viewer.GetNode(node_index[index]);
-      for (const auto* node_arg : node->OutputDefs()) {
-        if (node_arg->Exists()) {
-          hash_str(node_arg->Name());
-        }
+  // hashing output of each node
+  const int number_of_ort_nodes = graph_viewer.NumberOfNodes();
+  std::vector<size_t> nodes_vector(number_of_ort_nodes);
+  std::iota(std::begin(nodes_vector), std::end(nodes_vector), 0);
+  const std::vector<NodeIndex>& node_index = graph_viewer.GetNodesInTopologicalOrder();
+  for (const auto& index : nodes_vector) {
+    const auto& node = graph_viewer.GetNode(node_index[index]);
+    for (const auto* node_arg : node->OutputDefs()) {
+      if (node_arg->Exists()) {
+        hash_str(node_arg->Name());
       }
     }
+  }
 
 #ifdef __linux__
-    hash_str("LINUX");
+  hash_str("LINUX");
 #elif defined(_WIN32)
-    hash_str("WINDOWS");
+  hash_str("WINDOWS");
 #endif
 
 #ifdef ORT_VERSION
-    hash_str(ORT_VERSION);
+  hash_str(ORT_VERSION);
 #endif
 
 #ifdef CUDA_VERSION
-    hash_str(std::to_string(CUDA_VERSION));
+  hash_str(std::to_string(CUDA_VERSION));
 #endif
 
 #if defined(NV_TENSORRT_MAJOR) && defined(NV_TENSORRT_MINOR)
-    std::string TRT_VERSION = std::to_string(NV_TENSORRT_MAJOR) + "." + std::to_string(NV_TENSORRT_MINOR);
-    hash_str(TRT_VERSION);
+  std::string TRT_VERSION = std::to_string(NV_TENSORRT_MAJOR) + "." + std::to_string(NV_TENSORRT_MINOR);
+  hash_str(TRT_VERSION);
 #endif
 
-    model_hash = hash[0] | (uint64_t(hash[1]) << 32);
+  model_hash = hash[0] | (uint64_t(hash[1]) << 32);
 
-    // return the current unique id, and increment to update
-    return trt_model_id_[model_hash]++;
-  }
-
- private:
-  std::unordered_map<HashValue, int> trt_model_id_;       // current unique id for model
+  // return the current unique id, and increment to update
+  return trt_model_id_[model_hash]++;
 };
 
-std::unique_ptr<TRTModelIdGenerator> trt_model_id_generator_ = std::make_unique<TRTModelIdGenerator>();
-
 // Calll TRTGenerateModelId to generate hash id for TRT engine cache
-int TRTGenerateModelId(const GraphViewer& graph_viewer, HashValue& model_hash) {
+int TRTGenerateModelId(const GraphViewer& graph_viewer, HashValue& model_hash, std::unordered_map<HashValue, int>& trt_model_id_) {
```

Comment:
Do you need put trt_model_id_ in argument list? TRTGenerateModelId is a class method and can access class variable trt_model_id_.

Reply:
The reason bringing cache related method back to util.h is to prevent trt_basic_test from including TRTEP.h, which caused errors originally from NvInfer.h (this is included by TRTEP.h)

Since var trt_model_id_ is in class TRTEP and TRTGenerateModelId now becomes method in util.h, I am thinking if this design necessary in order for trt_basic_test to test this method without the need of class TRTEP

Reply:
you should be able to call TRTGenerateModelId() by instantiating TensorrtExecutionProvider class in your test just like what other tests did in tensorrt_basic_test.cc

Reply:
Do you mean to init TRTEP class like [this](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/test/providers/tensorrt/tensorrt_basic_test.cc#L408-L442)?